tests/Queries.elm

```diff
@@ -252,6 +253,21 @@ testChildren output =
         ]
 
 
+testHasChildWith : Single msg -> Test
+testHasChildWith output =
+    test "Selector.hasChildWith" <|
+        \() ->
+            output
+                |> Query.findAll
+                    [ tag "button"
+                    , hasChildWith [ text "click me" ]
+                    ]
+                |> Expect.all
+                    [ Query.count (Expect.equal 1)
+                    , Query.first >> Query.has [ class "super-button" ]
+                    ]
```

Comment:
Should we have some tests for what the failure messages look like?  I feel like there is the possibility of that getting complicated, especially if there are nested `hasChildWith`s